app/api/actions/reservations.js

```diff
@@ -11,6 +14,21 @@ function fetchReservations(params = {}) {
   });
 }
 
+function makeReservation(reservation) {
+  return createApiAction({
+    endpoint: 'reservation',
+    method: 'POST',
+    type: 'RESERVATION',
+    body: parseReservationData(reservation),
+  });
+}
+
+function parseReservationData(reservation) {
+  const parsed = pickBy(reservation, value => value);
```

Comment:
The API will throw an error if reservation is saved or edited with empty string values. Saving 0 values should be made possible so I'll change that at least. I'll have to test with null values but depending on the API that could be valid use case.